Saule/TypeExtensions.cs

```diff
@@ -33,5 +33,21 @@ public static IEnumerable<Type> GetInheritanceChain(this Type type)
                 .Concat(type.BaseType.GetInheritanceChain())
                 .Distinct();
         }
+
+        public static bool IsEnumerable(this Type type)
+        {
+            return type.IsGenericType && type.GetGenericTypeDefinition() == typeof(IEnumerable<>);
+        }
+
+        /// <summary>
+        /// Get the type of the items if the type is an <see cref="IEnumerable{T}"/> (if not it will return the input type).
+        /// </summary>
+        /// <param name="type">The type to apply this extension method to.</param>
+        /// <returns>The type of the items if the type is an <see cref="IEnumerable{T}"/> (if not it will return the input type).</returns>
+        public static Type TryGetCollectionType(this Type type)
```

Comment:
This name is a bit confusing; maybe `GetGenericTypeParameterOfCollection` is more clear?